File: Runtime/TransitionSpreadDistributionCollection.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'************************************************************************************

Imports System.Collections.ObjectModel

''' <summary>
''' Transition Spread Distribution collection
''' </summary>
''' <remarks></remarks>
Friend Class TransitionSpreadDistributionCollection
    Inherits KeyedCollection(Of Integer, TransitionSpreadDistribution)

    Protected Overrides Function GetKeyForItem(item As TransitionSpreadDistribution) As Integer
        Return item.TransitionSpreadDistributionId
    End Function

End Class

File: Runtime/TransitionSpreadDistributionCollection.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System.Collections.ObjectModel;

namespace SyncroSim.STSim
{
    internal class TransitionSpreadDistributionCollection : KeyedCollection<int, TransitionSpreadDistribution>
    {
        protected override int GetKeyForItem(TransitionSpreadDistribution item)
        {
            return item.TransitionSpreadDistributionId;
        }
    }
}
